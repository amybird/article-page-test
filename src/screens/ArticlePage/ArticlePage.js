import React, { Component } from 'react';
import styled from 'styled-components/macro';
import ArticleData from '../../data/article'
import shortId from 'shortid';
import HeroImage from './components/HeroImageWithText';
import Paragraph from '../../ui/Paragraph';
import Quote from '../../ui/Quote';
import StickyImageWithText from './components/StickyImageWithText';

const ARTICLE_COMPONENTS = { 
    HeroImage,
    Paragraph,
    StickyImageWithText,
    Quote
}

class ArticlePage extends Component {

    chooseComponentType (type, ...props) { 
        const Component = ARTICLE_COMPONENTS[type];
        return <Component {...props} key={shortId.generate()}/>;
    }
    
    render() {

        return(
            <Article>
                {ArticleData.content.map((item, index) => (
                    this.chooseComponentType(item.contentType, item.content)
                ))}
            </Article>
        )
    }
}

const Article = styled.div`
    padding-top:200px;
`;

export default ArticlePage;