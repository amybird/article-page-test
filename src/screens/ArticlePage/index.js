import React from 'react';
import { Article } from './styles';
import { data } from '../../data/article';
import HeroImage from '../../components/HeroImageWithText';
import StickyImageWithText from '../../components/StickyImageWithText';

const articleData = data.content;

const ArticlePage = () => (
  <Article>
    <HeroImage data={articleData[0]} />
    <StickyImageWithText data={articleData[1]} />
    <StickyImageWithText data={articleData[2]} />
  </Article>
);

export default ArticlePage;
