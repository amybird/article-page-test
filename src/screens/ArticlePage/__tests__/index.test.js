import React from 'react';
import renderer from 'react-test-renderer';
import ArticlePage from '../index';

describe('Article Page', () => {
  it('should render a container', () => {
    const tree = renderer
      .create(<ArticlePage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
