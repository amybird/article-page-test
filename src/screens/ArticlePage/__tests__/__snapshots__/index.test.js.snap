// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Page should render a container 1`] = `
.c0 {
  position: relative;
}

.c0 .fade-in {
  visibility: visible;
  opacity: 1;
  -webkit-transition: opacity 0.3s linear;
  transition: opacity 0.3s linear;
}

.c0 .fade-out {
  visibility: hidden;
  opacity: 0;
  -webkit-transition: visibility 0s 0.3s,opacity 0.3s linear;
  transition: visibility 0s 0.3s,opacity 0.3s linear;
}

.c2 {
  background-color: rgba(0,0,0,0.6);
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
}

.c3 {
  position: absolute;
  color: white;
  width: 100%;
  top: 50%;
  padding: 0 30px;
}

.c3 * {
  font-weight: lighter;
}

.c3 span {
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 10px;
}

.c3 h1 {
  margin: 0;
  text-transform: uppercase;
  font-family: serif;
  font-size: 30px;
}

.c3 h3 {
  margin: 0;
  text-align: right;
  font-family: serif;
}

.c6 {
  font-family: serif;
  line-height: 26px;
  font-size: 20px;
  margin: 20px 0;
}

.c6::first-letter {
  font-size: 50px;
}

.c7 {
  font-family: serif;
  line-height: 26px;
  font-size: 20px;
  margin: 20px 0;
}

.c7::first-letter {
  font-size: 20px;
}

.c4 {
  width: 100%;
  max-width: 1080px;
  margin: 0 auto;
}

.c8 {
  font-size: 30px;
  line-height: 40px;
  width: 100%;
  padding: 10px;
  text-align: left;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9 {
  width: 100%;
  top: 5px;
  display: inline-block;
}

.c5 {
  width: 100%;
  display: inline-block;
  padding: 20px 10px;
}

@media (min-width:768px) {
  .c0 {
    height: 500px;
    overflow: hidden;
  }
}

@media (min-width:768px) {
  .c1 {
    position: absolute;
  }
}

@media (min-width:768px) {
  .c3 {
    padding: 0 0 0;
  }

  .c3 div {
    max-width: 1080px;
    margin: 0 auto;
    height: 100%;
  }

  .c3 span {
    font-size: 16px;
  }

  .c3 h1 {
    font-size: 70px;
    -webkit-letter-spacing: 2px;
    -moz-letter-spacing: 2px;
    -ms-letter-spacing: 2px;
    letter-spacing: 2px;
  }

  .c3 h3 {
    font-size: 26px;
  }
}

@media (min-width:768px) {
  .c8 {
    font-size: 40px;
    line-height: 50px;
  }
}

@media (min-width:768px) {
  .c9 {
    margin-top: 70px;
    padding: 20px;
    position: -webkit-sticky;
    position: sticky;
    width: 50%;
    vertical-align: top;
  }
}

@media (min-width:768px) {
  .c5 {
    width: 50%;
  }
}

<div
  className=""
>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://cache.net-a-porter.com/content/images/story-head-content-SUBBED-1556275864012.jpeg/w1900_q65.jpeg"
        />
        <img
          alt="this is a lovely image"
          src="https://cache.net-a-porter.com/content/images/story-head-content-SUBBED-1556275864012.jpeg/w600_q65.jpeg"
        />
      </picture>
    </div>
    <div
      className="c2 fade-in"
    />
    <div
      className="c3"
    >
      <div
        className="fade-in"
      >
        <span>
          Cover Story
        </span>
        <h1>
          Standing strong
        </h1>
        <h3>
          <i>
            with
          </i>
        </h3>
        <h3>
          Danai Gurira
        </h3>
      </div>
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <p
        className="c6"
      >
        Going unnoticed is not in Danai Gurira’s repertoire. When the 41-year-old actress walks into the roof restaurant of the Dream hotel in Hollywood, wearing faded denim, a cream silk blouse and mirrored aviators, she practically gleams in the morning light. It’s hard to say if the other patrons recognize her as General Okoye, the fierce protector of the Wakanda royal family in Black Panther, or the equally imposing zombie-slayer Michonne from The Walking Dead, or are just struck by the way she carries herself, but all eyes are on her.
      </p>
      <p
        className="c7"
      >
        “I’m running a writers’ room near here,” Gurira says, sitting down and peering over the railing at the sprawl of surrounding offices and studios. She is talking about the newest line on her résumé: head writer for a television mini-series. Aside from her commanding performances on screen, she’s also a feted writer whose plays have garnered multiple awards, including a Tony nomination. Now she’s adapting Chimamanda Ngozi Adichie’s acclaimed novel Americanah – a love story following a Nigerian couple through their complicated lives – with her longtime friend Lupita Nyong’o. The two became close in 2015 when Nyong’o starred in Gurira’s all-female play Eclipsed, about the Liberian civil war.
      </p>
      <p
        className="c8"
      >
        “It’s about collaborating NOT cooperating. I’m pretty willing to have a CLEAR conversation about what I think WORKS, and what doesn’t, and why”
      </p>
    </div>
    <div
      className="c9"
    >
      <picture>
        <img
          alt="this is a lovely image"
          src="https://cache.net-a-porter.com/content/images/story-body-content-SUBBED-0-1556266075382.jpeg/w950_q65.jpeg"
        />
      </picture>
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c9"
    >
      <picture>
        <img
          alt="this is a lovely image"
          src="https://cache.net-a-porter.com/content/images/story-body-content-SUBBED-0-1556276906399.jpeg/w950_q65.jpeg"
        />
      </picture>
    </div>
    <div
      className="c5"
    >
      <p
        className="c6"
      >
        Gurira is tight-lipped about her upcoming departure from The Walking Dead, for which she’s scheduled to guest star in a handful of episodes in the tenth season. “We start shooting soon, so I’ll have to start fitting in my workouts,” she says of her Facetime sessions with her trainer. In a month, she’ll return to Atlanta, now practically her third home – as the location for the long-running zombie series as well as the Marvel shoots – alongside her apartment in Brooklyn and home in Hollywood. “I have a really beautiful family in The Walking Dead,” is what she tells me. “We have pot luck dinners. We go to each other’s homes. We check in with each other. I also talk a lot to my show runner, Scott Gimple, who is kind of my ‘Bat phone’ as I navigate running a writer’s room.”
      </p>
      <p
        className="c7"
      >
        For the task of adapting Americanah, Gurira is actually uniquely qualified. “Funnily, me and Chimamanda have these interesting parallels,” she says. “We were both raised by academics, either near or on an African university campus. We saw our countries go through various things when we were growing up. And we came back to America for school. Lupita, too.”
      </p>
      <p
        className="c7"
      >
        Born to African parents in the US, Gurira returned to her family’s native Zimbabwe when she was five, and then came back to the States for college in Minnesota. “I am definitely a strong meshing of both worlds,” she says, explaining that she feels American and Zimbabwean in equal parts. She first ‘noticed’ her cultural composition when she was a student at Macalester College in Saint Paul, Minnesota. “I knew when I got here that I had to be vigilant. Being in a land of such access and excess, you have to really be clear about what you’re doing and why, and how you’re going to be responsible with all of the opportunities available to you as an American citizen. And it was clear to me that I was going to retain a tangible and active connection to Zimbabwe. When I moved to Africa as a girl, I thought I was American. Then when I came back, I realized how African I was.”
      </p>
      <p
        className="c8"
      >
        “I am a strong meshing of BOTH worlds. When I moved to Africa as a girl, I thought I was AMERICAN. Then when I came back, I realized how AFRICAN I was”
      </p>
    </div>
  </div>
</div>
`;
